### PR TITLE
let `make commands` work for pure objective-c files

### PR DESCRIPTION
**Pull request by Complexicon**
What does this implement/fix? Explain your changes.
---------------------------------------------------
this allows for creating a compile-commands.json for pure objective-c files aka .m files
which in turn allows for intellisense for these .m files

Does this close any currently open issues?
------------------------------------------
not that i am aware of

Any relevant logs, error output, etc?
-------------------------------------

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Operating System:** Debian Bookworm 12

**Platform:** 

**Target Platform:** iphone/tweak

**Toolchain Version:** 13.0

**SDK Version:** iPhone 16.5

### CODE REVIEW

File: makefiles/instance/rules.mk [hunk 1]
```diff
@@ -169,6 +169,11 @@ _TARGET_CXX := $(TARGET_CXX)
 ifneq ($(firstword $(_TARGET_CXX)),$(THEOS_BIN_PATH)/gen-commands.pl)
 TARGET_CXX = $(THEOS_BIN_PATH)/gen-commands.pl $(_THEOS_TMP_COMPILE_COMMANDS_FILE) $${PWD} cxx $(filter $(__USER_FILES),$<) -- $(_TARGET_CXX)
 endif
+# add missing objective-c hook for compile-commands.json
```
Comment (kabiroberai) [inline makefiles/instance/rules.mk:172]: let’s remove this comment, isn’t needed once this code is upstreamed

**Review by kabiroberai — COMMENTED** [review]
thanks! LGTM in principle, just one nit

File: makefiles/instance/rules.mk (hunk 1 above)
Comment (Complexicon) [reply makefiles/instance/rules.mk:172]: gotcha, will do really quick

File: makefiles/instance/rules.mk [hunk 2]
```diff
@@ -169,6 +169,10 @@ _TARGET_CXX := $(TARGET_CXX)
 ifneq ($(firstword $(_TARGET_CXX)),$(THEOS_BIN_PATH)/gen-commands.pl)
 TARGET_CXX = $(THEOS_BIN_PATH)/gen-commands.pl $(_THEOS_TMP_COMPILE_COMMANDS_FILE) $${PWD} cxx $(filter $(__USER_FILES),$<) -- $(_TARGET_CXX)
 endif
+_TARGET_CC := $(TARGET_CC)
+ifneq ($(firstword $(_TARGET_CC)),$(THEOS_BIN_PATH)/gen-commands.pl)
+TARGET_CC = $(THEOS_BIN_PATH)/gen-commands.pl $(_THEOS_TMP_COMPILE_COMMANDS_FILE) $${PWD} cxx $(filter $(__USER_FILES),$<) -- $(_TARGET_CC)
```
Comment (L1ghtmann) [inline makefiles/instance/rules.mk:174]: Looks good! Have one minor thing for consistency  
(has no impact on the end result currently):

```suggestion
TARGET_CC = $(THEOS_BIN_PATH)/gen-commands.pl $(_THEOS_TMP_COMPILE_COMMANDS_FILE) $${PWD} cc $(filter $(__USER_FILES),$<) -- $(_TARGET_CC)
```

**Review by L1ghtmann — CHANGES_REQUESTED** [review]

**Review by kabiroberai — APPROVED** [review]
should be good to merge after @L1ghtmann's request

**Comment by Complexicon** [comment]
can we please get this upstream?

**Comment by Luki120** [comment]
Can you like, be patient? You already requested a review, don’t push people that work on this for free and on their free time after a work/school day

**Review by L1ghtmann — APPROVED** [review]
Thanks again!

**Pull request merged**